src/main/java/adris/altoclef/tasks/container/LootContainerTask.java

```diff
@@ -57,34 +61,49 @@ protected Task onTick(AltoClef mod) {
         }
         Optional<Slot> optimal = getAMatchingSlot(mod);
         if (optimal.isEmpty()) {
+            _weDoneHere = true;
             return null;
         }
-        setDebugState("Looting a container for all of their " + target.toString());
+        setDebugState("Looting items: " + targets);
         return new ClickSlotTask(optimal.get());
     }
 
     @Override
     protected void onStop(AltoClef mod, Task task) {
-        if (!_wasProtected) {
-            mod.getBehaviour().removeProtectedItems(target);
+        if (ContainerType.screenHandlerMatches(ContainerType.CHEST)) StorageHelper.closeScreen();
+        for (Item item : _protected) {
+            mod.getBehaviour().removeProtectedItems(item);
         }
         mod.getBehaviour().pop();
     }
 
     @Override
     protected boolean isEqual(Task other) {
-        return other instanceof LootContainerTask && target == ((LootContainerTask) other).target;
+        return other instanceof LootContainerTask && targets == ((LootContainerTask) other).targets;
     }
 
     private Optional<Slot> getAMatchingSlot(AltoClef mod) {
-        List<Slot> slots = mod.getItemStorage().getSlotsWithItemContainer(target);
-        if (slots.isEmpty()) return Optional.empty();
-        else return Optional.ofNullable(slots.get(0));
+        for (Item item : targets) {
```

Comment:
What if we want curse of binding items? I'd add an argument in the constructor called `Predicate<ItemStack>` that defaults to `stack -> true`. When calling this task you can pass a predicate to return false when it has curse of binding.

Reply:
Oh true